src/validation.h

```diff
@@ -718,7 +718,8 @@ class CChainState
     bool AcceptBlock(const std::shared_ptr<const CBlock>& pblock, BlockValidationState& state, CBlockIndex** ppindex, bool fRequested, const FlatFilePos* dbp, bool* fNewBlock) EXCLUSIVE_LOCKS_REQUIRED(cs_main);
 
     // Block (dis)connection on a given view:
-    DisconnectResult DisconnectBlock(const CBlock& block, const CBlockIndex* pindex, CCoinsViewCache& view);
+    DisconnectResult DisconnectBlock(const CBlock& block, const CBlockIndex* pindex, CCoinsViewCache& view)
+        EXCLUSIVE_LOCKS_REQUIRED(::cs_main);
```

Comment:
Why use the `::cs_main` global reference here rather than `cs_main`?

Reply:
I think for new code we prefer `::global`, unless the name of the global is prefixed with `g_...`, in which case `::g_...` and `g_...` are allowed.

Reply:
Thanks.